spec/lib/process_settings/settings_spec.rb

```diff
@@ -5,25 +5,38 @@
 require 'process_settings/settings'
 
 describe ProcessSettings::Settings do
-  describe "initialize" do
-    it "should take a json doc" do
-      process_settings = described_class.new("carrier" => "AT&T")
+  let(:settings_json_doc) { { "log_level" => "debug" } }
+  subject { described_class.new(settings_json_doc) }
 
-      expect(process_settings.json_doc.mine("carrier")).to eq("AT&T")
+  describe "#initialize" do
+    context "with a non-hash argument" do
+      let(:settings_json_doc) { "{}" }
+
+      it "raises" do
+        expect { subject }.to raise_exception(ArgumentError, /Settings must be a Hash/)
+      end
+    end
+
+    context "with a symbol key" do
+      let(:settings_json_doc) { { "gem" => { log_level: "debug" } } }
+
+      it "raises" do
+        expect { subject }.to raise_exception(ArgumentError, /symbol key :log_level found/)
+      end
     end
 
-    it "should reject anything not a hash" do
-      expect do
-        described_class.new("{}")
-      end.to raise_exception(ArgumentError, /Settings must be a Hash/)
+    context "with a symbol key" do
+      let(:settings_json_doc) { { "log_level" => :debug } }
+
+      it "raises" do
+        expect { subject }.to raise_exception(ArgumentError, /symbol value :debug found/)
+      end
     end
   end
 
   describe "#json_doc" do
-    it "should return what was stored" do
-      process_settings = described_class.new("carrier" => "AT&T")
-
-      expect(process_settings.json_doc).to eq("carrier" => "AT&T")
+    it "returns what was stored" do
```

Comment:
If you change lines 8-9 to be:
```
let(:settings_json_doc) { { "log_level" => "debug" } }
let(:settings) { described_class.new(settings_json_doc) }
subject { settings }
```

You can then make this:
```
subject { settings.json_doc }
it { should eq(settings_json_doc }
```

Doing the first change has generally been beneficial in my writing of RSpec tests because in the cases where I want to change the subject to a different initialization, I can redefine the `let` and the subject carries through.